### PR TITLE
completion: bash: misc fixes and updates

### PR DESCRIPTION
**Pull request by casperdcl**
- [x] add `metrics diff`
- [x] add DVC-file completion for `status`
- [x] fix `metrics show`
- [x] correct typo in `metrics remove` <= `metrics rmeove`
- [x] correct DVC-file completion for `lock`
- [x] correct file completion for `metrics remove`
- [x] correct file completion for `move`
- [x] remove deprecated options
  + [x] `run -c --cwd -y --yes --remove-outs`
- alphabetically sort options
- related: #2985

### CODE REVIEW

File: scripts/completion/dvc.bash [hunk 1]
```diff
@@ -24,26 +24,26 @@ _dvc_destroy='-f --force'
 _dvc_diff='-t --show-json --show-hash'
 _dvc_fetch='-j --jobs -r --remote -a --all-branches -T --all-tags -d --with-deps -R --recursive'
 _dvc_fetch_COMPGEN=_dvc_compgen_DVCFiles
-_dvc_get_url=''
+_dvc_gc='-a --all-branches -T --all-tags -w --workspace -c --cloud -r --remote -f --force -p --projects -j --jobs'
```
Comment (skshetry) [inline scripts/completion/dvc.bash]: `--all-commits` was hidden before. But, #3429 made it visible.

Comment (skshetry) [reply scripts/completion/dvc.bash]: Oh I see that issue now. It is not deprecated, just hidden.

File: scripts/completion/dvc.bash [hunk 2]
```diff
@@ -24,26 +24,26 @@ _dvc_destroy='-f --force'
...
 _dvc_metrics_add_COMPGEN=_dvc_compgen_files
-_dvc_metrics_show='$(-t --type -x --xpath -a --all-branches -T --all-tags -R --recursive'
-_dvc_metrics_show_COMPGEN=_dvc_compgen_files
 _dvc_metrics_diff='--targets -t --type -x --xpath -R --show-json'
 _dvc_metrics_modify='-t --type -x --xpath'
 _dvc_metrics_modify_COMPGEN=_dvc_compgen_files
-_dvc_metrics_remove='$(compgen -f)'
-_dvc_move='$(compgen -f)'
+_dvc_metrics_remove=_dvc_compgen_files
+_dvc_metrics_show='$(-t --type -x --xpath -a --all-branches -T --all-tags -R --recursive'
```
Comment (skshetry) [inline scripts/completion/dvc.bash]: Does this not require matching parens? :confused:

File: scripts/completion/dvc.bash (hunk 1 above)
Comment (casperdcl) [reply scripts/completion/dvc.bash]: #2681

File: scripts/completion/dvc.bash (hunk 2 above)
Comment (casperdcl) [reply scripts/completion/dvc.bash]: not really; requires no parens

File: scripts/completion/dvc.bash (hunk 1 above)
Comment (skshetry) [reply scripts/completion/dvc.bash]: @casperdcl, `--all-commits` is now visible and supported. Please check #3429.

Comment (casperdcl) [reply scripts/completion/dvc.bash]: Ah I see; didn't realise as it's not in the latest release. Right.

**Pull request merged**